includes/class-wc-payments.php

```diff
@@ -1673,6 +1674,40 @@ public static function enqueue_dev_runtime_scripts() {
 		}
 	}
 
+	/**
+	 * Adds WCPay options to Woo Core option allow list.
+	 *
+	 * @param   array $permissions Array containing the permissions.
+	 *
+	 * @return  array              An array containing the modified permissions.
+	 */
+	public static function add_wcpay_options_to_woocommerce_permissions_list( $permissions ) {
+		$wcpay_permissions_list = array_fill_keys(
+			[
+				'wcpay_frt_discover_banner_settings',
+				'wcpay_multi_currency_setup_completed',
+				'woocommerce_dismissed_todo_tasks',
+				'woocommerce_remind_me_later_todo_tasks',
+				'woocommerce_deleted_todo_tasks',
+				'wcpay_fraud_protection_welcome_tour_dismissed',
+				'wcpay_capability_request_dismissed_notices',
+				'wcpay_onboarding_eligibility_modal_dismissed',
+			],
+			true
+		);
+
+		if ( is_array( $permissions ) ) {
```

Comment:
**Non-blocking:** What do you think about simplifying the logic? I think we can remove the `$permissions` variable without compromising readability.

```suggestion
		if ( is_array( $permissions ) ) {
			return array_merge(
				$permissions,
				$wcpay_permissions_list
			);
		}
		
		return $wcpay_permissions_list;
```

Reply:
I like it. I agree that it doesn't compromise readability. I've implemented it in https://github.com/Automattic/woocommerce-payments/pull/7483/commits/cd207d1c6677155aedb511089a175bf4f359d04d.

Reply:
I also meant to mention that I re-tested it and everything works as expected.